properties/basic.py

```diff
@@ -573,19 +573,27 @@ def choices(self):
 
     @choices.setter
     def choices(self, value):
-        if not isinstance(value, (list, tuple, dict)):
-            raise TypeError('value must be a list, tuple, or dict')
-        if isinstance(value, (list, tuple)):
-            value = {v: v for v in value}
+        if isinstance(value, (set, list, tuple)):
+            if len(value) != len(set(value)):
+                raise TypeError("'choices' must contain no duplicate strings")
+            value = {v: [] for v in value}
+        if not isinstance(value, dict):
+            raise TypeError("'choices' must be a set, list, tuple, or dict")
         for key, val in value.items():
-            if not isinstance(val, (list, tuple)):
+            if isinstance(val, (set, list, tuple)):
+                value[key] = list(val)
+            else:
                 value[key] = [val]
+        all_items = []
         for key, val in value.items():
             if not isinstance(key, string_types):
-                raise TypeError('value must be strings')
+                raise TypeError("'choices' must be strings'")
             for sub_val in val:
                 if not isinstance(sub_val, string_types):
-                    raise TypeError('value must be strings')
+                    raise TypeError("'choices' must be strings")
+            all_items = all_items + [key] + val
```

Comment:
I think `all_items += [key] + val ` is easier to read and faster (does not matter too much here)